File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.py
def solution(numbers):
    # 영어 숫자와 실제 숫자의 매핑 리스트 생성
    words = ["zero", "one", "two", "three", "four", 
             "five", "six", "seven", "eight", "nine"]
    
    # 주어진 문자열(numbers)에서 영어 숫자를 실제 숫자로 변환
    for i, v in enumerate(words):
        numbers = numbers.replace(v, str(i))
    
    # 최종적으로 변환된 문자열을 정수로 변환하여 반환
    return int(numbers)

# ✅ 예제 테스트 실행
print(solution("onetwothreefourfivesixseveneightnine"))  
# 결과: 123456789 (영어 숫자가 숫자로 변환됨)

print(solution("onefourzerosixseven"))  
# 결과: 14067 (영어 숫자가 숫자로 변환됨)

File: 프로그래머스/0/120894. 영어가 싫어요/영어가 싫어요.java
public class Solution {

    public static long solution(String numbers) {
        String[] words = {
            "zero", "one", "two", "three", "four",
            "five", "six", "seven", "eight", "nine"
        };

        for (int i = 0; i < words.length; i++) {
            numbers = numbers.replace(words[i], String.valueOf(i));
        }

        return Long.parseLong(numbers);  // ← 변경: long 처리
    }

    public static void main(String[] args) {
        System.out.println(solution("onetwothreefourfivesixseveneightnine")); // 123456789
        System.out.println(solution("nineninenineninenine")); // 99999
        System.out.println(solution("nine" + "nine".repeat(10))); // 매우 큰 수
    }
}
